packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -450,108 +351,63 @@ export const InputBoxComponent = (props: InputBoxComponentProps): JSX.Element =>
+        const { updatedStartIndex, updatedEndIndex, shouldSetTargetSelectionUndefined } =
+          getSelectionIndicesWhenSelectionChangedByMouseDown(
+            event,
+            inputTextValue,
+            tags,
+            selectionStartValue,
+            selectionEndValue,
+            targetSelection
+          );
+        setSelectionStartValue(updatedStartIndex);
+        setSelectionEndValue(updatedEndIndex);
+        if (shouldSetTargetSelectionUndefined) {
           setTargetSelection(undefined);
-        } else if (event.currentTarget.selectionStart !== null) {
-          // on select was triggered by mouse down/up with no movement
-          const mentionTag = findMentionTagForSelection(tags, event.currentTarget.selectionStart);
-          if (mentionTag !== undefined && mentionTag.plainTextBeginIndex !== undefined) {
-            // handle mention click
-            // Get range of word that was clicked on
-            const selectionRange = rangeOfWordInSelection({
-              textInput: inputTextValue,
-              selectionStart: event.currentTarget.selectionStart,
-              selectionEnd: event.currentTarget.selectionEnd,
-              tag: mentionTag
-            });
-
-            if (event.currentTarget.selectionDirection === null) {
-              event.currentTarget.setSelectionRange(selectionRange.start, selectionRange.end);
-            } else {
-              event.currentTarget.setSelectionRange(
-                selectionRange.start,
-                selectionRange.end,
-                event.currentTarget.selectionDirection
-              );
-            }
-            setSelectionStartValue(selectionRange.start);
-            setSelectionEndValue(selectionRange.end);
-          } else {
-            setSelectionStartValue(event.currentTarget.selectionStart);
-            setSelectionEndValue(event.currentTarget.selectionEnd);
-          }
         }
       } else {
         // selection was changed by keyboard
-        updateSelectionIndexesWithMentionIfNeeded(event, inputTextValue, selectionStartValue, selectionEndValue, tags);
+        const { updatedStartIndex, updatedEndIndex } = updateSelectionIndicesWhenSelectionChangedByKeyboard(
+          event,
+          inputTextValue,
+          selectionStartValue,
+          selectionEndValue,
+          tags
+        );
+        setSelectionStartValue(updatedStartIndex);
+        setSelectionEndValue(updatedEndIndex);
       }
       // don't set setShouldHandleOnMouseDownDuringSelect(false) here as setSelectionRange
       // could trigger additional calls of onSelect event and they may not be handled correctly
       // (because of setSelectionRange calls or rerender)
     },
-    [
-      updateSelectionIndexesWithMentionIfNeeded,
-      targetSelection,
-      setTargetSelection,
-      setSelectionStartValue,
-      setSelectionEndValue
-    ]
+    [caretIndex]
```

Comment:
`setSelectionStartValue` and `setSelectionEndValue` are called above, so why have they been removed from the array here?